### PR TITLE
Normalize version when applying deletes

### PR DESCRIPTION
**Pull request by joelverhagen**
Fix https://github.com/NuGet/Engineering/issues/399

### CODE REVIEW

File: src/Catalog/Registration/RegistrationKey.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public override bool Equals(object obj)
                 return false;
             }
 
-            return (Id == rhs.Id); 
+            return StringComparer.OrdinalIgnoreCase.Equals(Id, rhs.Id);
```
Comment (skofman1) [inline src/Catalog/Registration/RegistrationKey.cs]: we need tests for this class that make sure that if 2 objects are equal with Equals, GetHashCode also equals

File: src/Catalog/Registration/RegistrationEntryKey.cs [hunk 2]
```diff
@@ -21,7 +26,7 @@ public override string ToString()
 
         public override int GetHashCode()
         {
-            return ToString().GetHashCode();
+            return $"{RegistrationKey}/{_normalizedVersion}".GetHashCode();
```
Comment (skofman1) [inline src/Catalog/Registration/RegistrationEntryKey.cs]: same here

File: src/Catalog/Registration/RegistrationEntryKey.cs [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
 {
     public class RegistrationEntryKey
     {
+        private readonly string _normalizedVersion;
+
         public RegistrationEntryKey(RegistrationKey registrationKey, string version)
         {
             RegistrationKey = registrationKey;
             Version = version;
+            _normalizedVersion = NuGetVersionUtility.NormalizeVersion(version).ToLowerInvariant();
```
Comment (skofman1) [inline src/Catalog/Registration/RegistrationEntryKey.cs]: I wonder if ToLowerInvariant should go into a more common location, so all the code converts a version  to string in the same way

Comment (joelverhagen) [inline src/Catalog/Registration/RegistrationEntryKey.cs]: In general, we've had to make these sort of changes in a localized way. For example, JSON blobs should always contain the original casing but URLs should always contain the normalized + lower-cased version.

File: src/Catalog/Registration/RegistrationEntryKey.cs (hunk 2 above)
Comment (joelverhagen) [inline src/Catalog/Registration/RegistrationEntryKey.cs]: Fixed.

File: src/Catalog/Registration/RegistrationKey.cs (hunk 1 above)
Comment (joelverhagen) [inline src/Catalog/Registration/RegistrationKey.cs]: Fixed.

**Pull request merged**